lib/data/wrapper.js

```diff
@@ -256,6 +257,19 @@ const data = {
                 log.error('delete error from datastore',
                     { error: err, key: objectGetInfo.key, moreRetries: 'no' });
             }
+            if (!isMdModelVersion2) {
+                // pass size as negative so location metric is decremented
+                return pushLocationMetric(objectGetInfo.dataStoreName,
```

Comment:
You will want to pass the decrement when `err` is not null (for the `ObjNotFound` case)